pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -484,3 +561,41 @@ func newCondition(status corev1.ConditionStatus, reason string, currentTime meta
 
 	return condition
 }
+
+func getHorizontalEventsRetention(policy *datadoghq.DatadogPodAutoscalerPolicy, longestLookbackAllowed time.Duration) time.Duration {
+	var longestRetention time.Duration
+	if policy == nil {
+		return longestRetention
+	}
+
+	if policy.Upscale != nil {
+		upscaleRetention := getLongestScalingRulesPeriod(policy.Upscale.Rules)
+		if upscaleRetention > longestRetention {
+			longestRetention = upscaleRetention
+		}
```

Comment:
```suggestion
		longestRetention = getLongestScalingRulesPeriod(policy.Upscale.Rules)
```
I think `upscaleRetention > longestRetention` will either be true or equal unless we allow negative values